class/defaults.yml

```
@@ -1,16 +1,17 @@
 parameters:
+  multi_instance: true
   keycloak:
     namespace: syn-keycloak
-    release_name: keycloak
+    release_name: keycloak-${_instance}
     charts:
       keycloak: '10.3.1'
     # FQDN should be overwritten on the cluster level
     fqdn: keycloak.example.com
     # Keycloak Admin
     admin:
-      secretname: keycloak-admin-user
+      secretname: ${keycloak:release_name}-admin-user
       username: admin
-      password: '?{vaultkv:${customer:name}/${cluster:name}/keycloak/admin-password}'
+      password: '?{vaultkv:${customer:name}/${cluster:name}/keycloak/${_instance}-admin-password}'
```

consider keeping each instance's secrets in a different vault secret:

```suggestion
      password: '?{vaultkv:${customer:name}/${cluster:name}/${_instance}/admin-password}'
```

---

Right. I didn't know the convention here. I was inspired by Minio: https://github.com/projectsyn/component-minio/blob/master/class/defaults.yml#L9